### DISCUSSION TITLE
The select of "Related product" shows disabled products

### DISCUSSION THREAD

**Discussion by Maofree** [PrestaShop/PrestaShop#40144] [Ideas]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [x] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

Hi
a my customer report me a problem on the select of "Related product"
in the website there are 4000 active products and other are disabled because the website has some years and normally some products have gone out of production and are being deactivated.

He has problems to select active Related products from the product selection, because he doesn't know which product is disabled.
I think the selector should only show active products, it is useless to choose products that have been deactivated on the site

can you add the condition to show only active products in that selector?

thanks
bye

### Expected behavior

Hi
in the related product selection should be present only active products

### Steps to reproduce

1. go in the admin panel, in the product list and disable a product
2. go in a product page and write the name of that disabled product
3. now you will see this product in the selection

look at this video
https://watch.screencastify.com/v/IlHPlC6MfhYaiFJh6npJ

### PrestaShop version(s) where the bug happened

1.7.8.6

### PHP version(s) where the bug happened

7.3

**Comment by AureRita** [comment]
Hi @maofree 

Thank you for your reporting. Currently, I think it's a planned feature. Indeed, if you want to planify some product to be related but you disabled some of them because they aren't ready to be sell, you can do it and when you'll enabled them, they'll be already related. You can see it on this video : 

https://user-images.githubusercontent.com/100699445/179193187-64fd4f01-9860-43cb-85a1-52933f2295cf.mp4

As you can see, the related product isn't showed on the FO ( the page for the client )

I'll close this issue because, for me, it's a planned feature, if you're not agree with this result, you can say it on this thread and I can re-open the ticket

**Comment by Maofree** [comment]
can you check if it planed?

the fact that they do not appear on the front I think is normal, because the products are deactivated.
what I would like is that they do not even appear in the admin side selection. I don't think it's a planned change
can you reopen this bug report?

**Comment by Maofree** [comment]
image a website with 5000 or more products
with 500 disabled products
in the selection of related product you see also the disabled products but without to know which are active or disabled.
I know that in the front office these related products if disabled will not be shown, but the problem is on the select not in the front-office

the fact that they are not shown on the front if disabled reinforces this request, what sense does it make to select deactivated products if they will not be displayed on the front?

**Comment by AureRita** [comment]
As you asked me, I checked it. Currently it's planed but it can be a good feature to implement, so I'll treat it as if it were one, 

Thank you for your suggestion. The Product Team will take it into consideration for future developments.

Please be aware that there is no guarantee that this feature will be developed anytime soon.

Thank you

**Comment by Maofree** [comment]
ok thanks
